src/main/java/StringCalculator.java

```diff
@@ -0,0 +1,81 @@
+import java.util.Arrays;
+
+public class StringCalculator {
+
+    private static int[] ExpressionToTokens(String expression) {
+        try {
+            String tokens = expression;
+            String delimiters = "[,:]";
+
+            if (expression.contains("//") && expression.contains("\n")) {
+                String customDelimiter = extractCustomDelimiter(expression);
+                tokens = extractToken(expression);
+                delimiters = "[,:" + customDelimiter + "]";
+            }
+
+            String[] splitTokens = tokens.split(delimiters);
+            return StringArrayToIntArray(splitTokens);
+        } catch (NumberFormatException e) {
+            throw new RuntimeException("Contains non-numerical values.");
+        }
+    }
+
+    private static String extractCustomDelimiter(String expression) {
+        int startIndex = expression.indexOf("//") + 2;
+        int endIndex = expression.indexOf("\n");
+        return expression.substring(startIndex, endIndex);
+    }
+
+    private static String extractToken(String expression) {
+        int startIndex = expression.indexOf("//");
+        int endIndex = expression.indexOf("\n");
+
+        return expression.substring(0, startIndex) + expression.substring(startIndex + 2, endIndex) + expression.substring(endIndex + 1);
```

Comment:
`expression.substring(0, startIndex)`는 항상 빈 문자열이라 불필요한 문자열 연결이 발생하고 있어요!
현재 목적이 숫자 부분만 추출하는 것이라면,

```suggestion
        int endIndex = expression.indexOf("\n");

        return expression.substring(endIndex + 1);
```

이렇게 수정하면 불필요한 연산이 줄어서 가독성도 높일 수 있을 거 같은데 어떻게 생각하세요?

Reply:
처음에는 저도 멘토님이 말씀하신 대로 expression.substring(endIndex + 1) 방식으로 구현을 시도했지만
해당 방식은 단순히 \n 이후의 문자열만 반환하기 때문에 커스텀 구분자를 추출하고 적용하는 과정이 생략되는 문제가 발생했습니다.
따라서, 이를 해결하기 위해 모든 경우에서 정상적으로 동작하도록 수정하였습니다!